packages/react/src/components/timeInput/TimeInput.tsx

```diff
@@ -0,0 +1,317 @@
+import React, { FocusEventHandler, useLayoutEffect, useRef, useState } from 'react';
+import 'hds-core';
+import isFunction from 'lodash.isfunction';
+
+import { InputWrapper } from '../../internal/input-wrapper/InputWrapper';
+import { TextInputProps } from '../textInput/TextInput';
+import textInputStyles from '../textInput/TextInput.module.css';
+import styles from './TimeInput.module.scss';
+import classNames from '../../utils/classNames';
+
+const NUMBER_KEYS: string[] = '0,1,2,3,4,5,6,7,8,9'.split(',');
+
+/**
+ * Pad a one-char string with a leading zero
+ */
+const zeroPad = (value: string) => {
+  if (value.length === 1) {
+    return `0${value}`;
+  }
+  return value;
+};
+
+/**
+ * Increment a number inside a range
+ * @param min Min number to return
+ * @param max Max number to return
+ * @param current Number to modify
+ * @param modifier Modifier number to add tot the current number
+ */
+const incrementNumber = (min: number, max: number, current: number, modifier: number) => {
+  return Math.max(Math.min(current + modifier, max), min);
+};
+
+/**
+ * Returns the default hour/minute values from a defaultValue string
+ */
+const getDefaultValues = (defaultValue?: string): string[] | null => {
+  const defaultValueString = `${defaultValue}`;
+  if (defaultValue && defaultValueString.length > 0) {
+    if (defaultValueString.match(/^\d{2}:\d{2}$/)) {
+      return defaultValueString.split(':');
+    }
+    // eslint-disable-next-line no-console
+    console.warn('Invalid default value for TimeInput. The default value must be in hh:mm format');
+  }
+  return null;
+};
+
+const isShortNumericString = (inputValue: string): boolean => inputValue.match(/^(\d{1,2})?$/) !== null;
+
+export const TimeInput = React.forwardRef<HTMLInputElement, TextInputProps>(
+  (
+    {
+      className = '',
+      disabled = false,
+      defaultValue,
+      errorText,
+      helperText,
+      hideLabel,
+      invalid,
+      id,
+      label,
+      labelText,
+      onChange = () => null,
+      required,
+      style,
+      successText,
+      tooltipLabel,
+      tooltipText,
+      tooltipButtonLabel,
+      type = 'text',
+      ...rest
+    }: TextInputProps,
+    ref?: React.Ref<HTMLInputElement>,
+  ) => {
+    const defaultValues = getDefaultValues(defaultValue);
+    const inputRef = useRef<HTMLInputElement>(null);
```

Comment:
Why do we need to create a new inputRef here, instead of applying the one that is provided through props?

Reply:
I guess this ref is for the hidden input.

Reply:
Yeah I wonder if we could check here if the ref is provided, and then apply it here directly instead of new ref? Then we might not need the stuff done at line 131 where props.ref is merged with the internal ref 🤔

Reply:
I'm not sure would it cause some problems 🤔

Reply:
It probably makes a code bit more complex because the we would need to know every time we use it if it is a Ref or MutableRefObject:
`inputRef || inputRef.current`

Reply:
It would be complex or even impossible because the ref prop can be a function returning HTML Element or a RefObject:
`RefCallback<T> | RefObject<T> | null;`